Repository/RepoTreeView.cpp

```diff
@@ -143,6 +146,64 @@ void RepoTreeView::onRepoDeactivated(RM::Repo* repo)
     }
 }
 
+/**
+ * @brief This slot is called, when a single fetch operation finishes.
+ */
+void RepoTreeView::fetchOperationFinished()
+{
+    Git::FetchOperation* op = qobject_cast<Git::FetchOperation*>( sender() );
+    Q_ASSERT( op );
+
+    Git::Result r( op->result() );
+    if ( !r ) {
+        QMessageBox::warning( this, tr("Operation failed."),
+                              tr("Failed to fetch repository '%1'.\nMessage: %2")
+                              .arg(op->repository().name()).arg(r.errorText())
+                              );
+    }
+
+    // delete the operation
+    op->deleteLater();
+}
+
+void RepoTreeView::onCtxFetchAll()
+{
+    Heaven::Action* action = qobject_cast< Heaven::Action* >( sender() );
```

Comment:
If we add a `return;` here, we could merge this code now and activate it when we're there...